subsys/mpsl/mpsl_init.c

```diff
@@ -155,7 +155,10 @@ static int mpsl_lib_init(const struct device *dev)
 	clock_cfg.skip_wait_lfclk_started =
 		IS_ENABLED(CONFIG_SYSTEM_CLOCK_NO_WAIT);
 
-#ifdef CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC_CALIBRATION
+#ifdef CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC
+#ifndef CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC_CALIBRATION
+#error "MPSL requires clock calibration to be enabled when RC is used as LFCLK"
+#endif
```

Comment:
```suggestion
#ifdef CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC
BUILD_ASSERT(CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC_CALIBRATION,
 "MPSL requires clock calibration to be enabled when RC is used as LFCLK");
#endif
```

Reply:
Used `BUILD_ASSERT(IS_ENABLED(` as CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC_CALIBRATION may not be defined